### ISSUE TITLE
Using subnet_filter broken in 2.5.3 (related to security groups changes)

### ISSUE THREAD

**Issue by llibicpep** [test-kitchen/kitchen-ec2#394]
If I use `subnet_filter` in my `.kitchen.yml` and not define anything about subnets, `kitchen-ec2` tries to create a new SG. I am getting the following error:

```
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [No default VPC for this user] on default-centos-7
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

D      ------Exception-------
D      Class: Kitchen::ActionFailed
D      Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [No default VPC for this user] on default-centos-7
D      ----------------------
D      ------Backtrace-------
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:183:in `report_errors'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:174:in `run_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command/action.rb:36:in `block in call'
D      /opt/chefdk/embedded/lib/ruby/2.4.0/benchmark.rb:293:in `measure'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command/action.rb:34:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/cli.rb:52:in `perform'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/cli.rb:193:in `block (2 levels) in <class:CLI>'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/bin/kitchen:13:in `block in <top (required)>'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/errors.rb:171:in `with_friendly_errors'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/bin/kitchen:13:in `<top (required)>'
D      /opt/chefdk/bin/kitchen:250:in `load'
D      /opt/chefdk/bin/kitchen:250:in `<main>'
D      ----End Backtrace-----
D      -Composite Exception--
D      Class: Kitchen::ActionFailed
D      Message: Failed to complete #create action: [No default VPC for this user] on default-centos-7
D      ----------------------
D      ------Backtrace-------
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/plugins/response_target.rb:21:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/request.rb:70:in `send_request'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/kitchen-ec2-2.2.1/lib/kitchen/driver/ec2.rb:738:in `create_security_group'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/kitchen-ec2-2.2.1/lib/kitchen/driver/ec2.rb:215:in `create'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `public_send'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `block in perform_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:536:in `synchronize_or_call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:498:in `block in action'
D      /opt/chefdk/embedded/lib/ruby/2.4.0/benchmark.rb:293:in `measure'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:497:in `action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `perform_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:379:in `create_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:368:in `block in transition_to'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:367:in `each'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:367:in `transition_to'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:135:in `converge'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:197:in `public_send'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:197:in `run_action_in_thread'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:169:in `block (2 levels) in run_action'
D      ----End Backtrace-----
D      ---Nested Exception---
D      Class: Kitchen::ActionFailed
D      Message: Failed to complete #create action: [No default VPC for this user]
D      ----------------------
D      ------Backtrace-------
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/plugins/raise_response_errors.rb:15:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:20:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/idempotency_token.rb:18:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/aws-sdk-core/plugins/param_converter.rb:20:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/plugins/response_target.rb:21:in `call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/request.rb:70:in `send_request'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/aws-sdk-core-2.11.9/lib/seahorse/client/base.rb:207:in `block (2 levels) in define_operation_methods'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/kitchen-ec2-2.2.1/lib/kitchen/driver/ec2.rb:738:in `create_security_group'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/kitchen-ec2-2.2.1/lib/kitchen/driver/ec2.rb:215:in `create'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `public_send'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `block in perform_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:536:in `synchronize_or_call'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:498:in `block in action'
D      /opt/chefdk/embedded/lib/ruby/2.4.0/benchmark.rb:293:in `measure'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:497:in `action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:469:in `perform_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:379:in `create_action'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:368:in `block in transition_to'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:367:in `each'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:367:in `transition_to'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/instance.rb:135:in `converge'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:197:in `public_send'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:197:in `run_action_in_thread'
D      /opt/chefdk/embedded/lib/ruby/gems/2.4.0/gems/test-kitchen-1.20.0/lib/kitchen/command.rb:169:in `block (2 levels) in run_action'
D      ----End Backtrace-----
```

I think it tries to create new SG in default VPC, and I don't have any VPC marked as default in my account. It should be using VPC the subnet found by `subnet_filter` belong to.

**Comment by coderanger** [comment]
Yeah, looks like we need an extra `elsif` here https://github.com/test-kitchen/kitchen-ec2/blob/4adaa2a297cb614218025948915527ca153c109f/lib/kitchen/driver/ec2.rb#L728

**Comment by llibicpep** [comment]
What was the previous behavior? I feel like the change wasn't backward compatible... and actually no way to configure it the way so it does things like it used to be.

**Comment by coderanger** [comment]
Indeed, it was a big structural change and has had some collateral damage, sorry :(

**Comment by llibicpep** [comment]
@coderanger could you please let me know if my PR makes sense.

**Comment by christek91** [comment]
Hi all, is there any chance that the fix for this could be merged and released (once conflicts are fixed? ). 

My team has recently run into this as well 😞

**Comment by llibicpep** [comment]
@tas50 I've updated the PR

**Comment by jeffbyrnes** [comment]
Looks like this was fixed in #413, which was released as [v3.2.0](https://github.com/test-kitchen/kitchen-ec2/releases/tag/v3.2.0).

Seems like this should be closed, in that case, right?
